smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MetadataPropagationTest.java

```diff
@@ -66,6 +68,48 @@ public void testFromKafkaToAppToKafka() {
         });
     }
 
+    @Test
+    public void testFromKafkaToAppToKafkaForwardKey() {
+        String topicIn = UUID.randomUUID().toString();
+        LinkedHashMap<String, String> messages = new LinkedHashMap<>();
+        usage.consumeStrings(topic, 10, 1, TimeUnit.MINUTES, null, messages::put);
+        runApplication(getKafkaSinkConfigForAppProcessingDataForwardKey(topic, topicIn), MyAppForwardingKey.class);
+
+        AtomicInteger count = new AtomicInteger();
+        usage.produceIntegers(10, null, () -> {
+            int c = count.getAndIncrement();
+            return new ProducerRecord<>(topicIn, String.valueOf(c), c);
+        });
+
+        await().until(() -> messages.size() >= 10);
+
+        assertThat(messages).containsKeys("0", "1", "2", "3", "4", "5", "6", "7", "8", "9");
+        assertThat(messages).containsValues("1", "2", "3", "4", "5", "6", "7", "8", "9", "10");
+    }
+
+    @Test
+    public void testFromKafkaToAppToKafkaForwardKeyReturningMessage() {
```

Comment:
Can you add a test verifying when the incoming key is `null`?

Reply:
Pushed the test